### DISCUSSION TITLE
Export to HTML

### DISCUSSION THREAD

**Discussion by kaaax0815** [BetaHuhn/ackee-report#45] [Ideas] [answered]
I dont know if their is a way to do so, because the email is a html document, but you need
1. a mail server
2. a mail account
3. to convert quoted printables to human readable

**Comment by BetaHuhn** [comment, marked as answer]
I'm not quite sure I understand what you mean. 

Are you asking for a feature where the ackee-report exports the same report which is sent via email directly as an html file?

**Reply by kaaax0815** [reply]
Indeed

**Reply by BetaHuhn** [reply]
What would be the use case for such a feature? What would you do with the html file?

**Reply by kaaax0815** [reply]
i saw that you create the html with a simple template, it woudnt be hard to save instead of send the content

**Reply by BetaHuhn** [reply]
It would be possible, but I want to understand what benefits exporting the report as a html file directly would have.

You can already export the report as JSON for other integrations or xml for rss feeds. What would you do with the html file?

**Reply by kaaax0815** [reply]
You have a nice overview of all services, and get a graphical difference between e. G. the last month. And more features are nice xd. I work on this, and create a pull request when done. You decide to merge or not 🙂

**Reply by BetaHuhn** [reply]
The reason why I am so curious is, you already have a nice overview in Ackee's actual dashboard. 

This tool is supposed to be an extension if you want to generate periodic reports and send them with different services (email, rss, etc.).

So what would be the benefit of using this tool to generate a static report of your sites using Ackee? Do you like this design/structure better than Ackee's dashboard, or do you want to create different reports for individual clients/projects?

In those cases I think it might be better to just use Ackee's dashboard or bug Ackee's author for a different overview or new feature 😄

**Reply by kaaax0815** [reply]
Yeah you right, it wouldnt make sense for cronjob etc. But making a overview page for everything would be for me way more difficult. But doing direct html over email would also be nice. I'll make the feature for me still, but don't upload it because your tool doesnt need this feature. Maybe I'll create a fork. But the html to email is also a thing to work on. Do you thing this would be sinnvoll?

**Reply by BetaHuhn** [reply]
> But the html to email is also a thing to work on

I don't quite get what you mean by that. The HTML generated by ackee-report is already being sent via email.

**Reply by BetaHuhn** [reply]
As Ackee currently doesn't support individual pages for each domain (which you could share to clients/public) I think an option to export a report as a HTML file would be a nice addition. You could then host the file via other static hosting services and share it with other people.

Is this something you wanted to achieve?

**Reply by kaaax0815** [reply]
Yes

**Reply by BetaHuhn** [reply]
Okay, I guess we could simply add a html command similarly to the rss or json commands, but use most of the logic from the email command. You could then generate a html report like this:

```
ackee-report html -d all -o index.html
```

**Reply by kaaax0815** [reply]
I did this the way you wrote, because it makes most sense. I currently have to problem with output and I have to copy the function from the email thing.

**Reply by BetaHuhn** [reply]
You should be able to just copy the [render](https://github.com/BetaHuhn/ackee-report/blob/01ab2e37d64ed948c0582d5f6b39c03cf1cfff04/src/service/email.js#L21) function from the email service.

**Comment by kaaax0815** [comment]
Yes I think so to. I just had to get off so I didn't have to time implement
this. Tomorrow I have more time. And we dont have timezone change lul

Maximilian Schiller <notifications@github.com> schrieb am Di., 9. März
2021, 21:08:

**Comment by kaaax0815** [comment]
@BetaHuhn This is offtopic, but how do i trigger a release with semantic-release? i set the defaultbranch to main, but it says `The commit should not trigger a release` but it should...

**Reply by BetaHuhn** [reply]
Your commit needs to contain a specific format for it to trigger a release. I think by default it uses [AngularJS commit message format](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-format).

**Reply by kaaax0815** [reply]
How do you trigger the release, because you also use semantic-release

**Reply by kaaax0815** [reply]
I use your github release workflow. [my repo](https://github.com/kaaax0815/nodejsanalytics)

**Reply by BetaHuhn** [reply]
You also need a [release config file](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#configuration-file) like [this](https://github.com/BetaHuhn/ackee-report/blob/master/release.config.js)

**Reply by kaaax0815** [reply]
[Repo](https://github.com/kaaax0815/nodejsanalytics) I did everything like you (because your good, i gave you credit) but it doesnt work. it just says that the commit doesnt trigger a release

**Reply by kaaax0815** [reply]
the default angular syntax triggers the release even tho i use your config @BetaHuhn

**Reply by BetaHuhn** [reply]
That's weird, it should work with emojis if you use my config.

**Reply by kaaax0815** [reply]
other issues fixed by adding my bot to collabs, im trying gitmojis tommorrow. they are just too great to miss them out

**Reply by kaaax0815** [reply]
@BetaHuhn Now it works, thanks for your config, I dont know what it was but now it just works. the only problem is that my user creates the new commit with [skip ci] and not my bot. but my bot does the release. how do i change this? in my workflow it is all set to my bot...

**Reply by BetaHuhn** [reply]
You have to make sure to specify the email address of the bot account and also use it's GitHub PAT.

**Reply by kaaax0815** [reply]
@BetaHuhn I am now using dependabot aswell, do i have to merge the pr by myself or does dependabot itself. i saw that it does your bot is this the release scheduler workflow?

**Reply by BetaHuhn** [reply]
The release scheduler workflow releases any dependency updates in the specified interval as they don't trigger a semantic-release by themselves. 

[Here's](https://github.com/BetaHuhn/github-files/blob/master/workflows/dependencies/dependabot.yml) the workflow I use to merge Dependabot PRs. Dependabot does not automatically merge PRs it creates. My workflow uses [this GitHub Action](https://github.com/koj-co/dependabot-pr-action) which runs on a schedule and merges any Dependabot PRs which pass your CI.

**Reply by kaaax0815** [reply]
@BetaHuhn You have many good github templates/repos. what does the release-scheduler do which also is in this repo, i read the description but i cant figure out what it does

**Reply by BetaHuhn** [reply]
> You have many good github templates/repos.

Thanks!

> What does the release-scheduler do?

The commit which merges a dependency update does not trigger a release, because I don't want to create a release for each dependency update right after it was merged. 

Instead the release-scheduler runs at a specified interval and creates a release for those dependency updates. By the time it runs (in my case every Monday) there have often been made multiple dependency updates which can be released together.

Is it clearer now?

**Reply by kaaax0815** [reply]
So every week from all of the dependecy updates a release gets made. Makes sense

**Reply by BetaHuhn** [reply]
Exactly.

**Reply by kaaax0815** [reply]
What I still dont understand is with the semantic bot. the reply on the pull requests that is included in the new release and the release are made by my bot, only the commit is made by me

**Reply by BetaHuhn** [reply]
Do you mean the `🔖 Release vx.x.x [skip ci]` commit?

**Reply by kaaax0815** [reply]
yes

**Reply by kaaax0815** [reply]
My repo is my first real nodejs project, because im not a nodejs programmer and i just cant get a good code coverage. i just dont understand how to test functions, because supertest is only for express itself. i have just 67%

**Reply by BetaHuhn** [reply]
I don't really have the time to take a closer look at your codebase, sorry, Normally you can import each function by itself and test it with different parameters using your test runner/framework (like mocha).

**Reply by kaaax0815** [reply]
I respect all your help, youre a great and good (programming wise) person! Keep it up! I have to thank you for helping me getting into nodejs
